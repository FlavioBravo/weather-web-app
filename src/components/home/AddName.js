import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddName = ({ setCities }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            // setCities(cities => [inputValue, ...cities]);
            setCities(cities => [inputValue]);
            setInputValue('');
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <h4>Filters</h4>
            <label>Name:</label>
            <p> {inputValue} </p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddName.propTypes = {
    setCities: PropTypes.func.isRequired
}