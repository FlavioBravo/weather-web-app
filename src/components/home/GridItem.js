import React from 'react';
import PropTypes from 'prop-types';
import { useHistory } from "react-router-dom";

export const GridItem = ({ id, name, state, country, coord }) => {

    const history = useHistory();

    const handleForecast = () => {
        history.push(`/forecast/${id}`);
    };

    return (
        <tr className="animate__animated animate__fadeIn" onClick={ handleForecast } >
            <td> { id } </td>
            <td> { name } </td>
            <td> { state } </td>
            <td> { country } </td>
            <td> { coord.lon } </td>
            <td> { coord.lat } </td>
        </tr>
    )
}

GridItem.propTypes = {
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    state: PropTypes.string.isRequired,
    country: PropTypes.string.isRequired,
    coord: PropTypes.object.isRequired,
}