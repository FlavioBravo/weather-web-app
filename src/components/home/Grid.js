import React from 'react';
import PropTypes from 'prop-types';

import { useFetchCities } from '../../hooks/useFetchCities';
import { GridItem } from './GridItem';

export const Grid = ({ city }) => {

    const { data: images, loading } = useFetchCities(city);

    return (
        <>
            <br />
            <h3 className="animate__animated animate__fadeIn"> {city} </h3>
            <br />
            { loading && <p className="animate__animated animate__flash">Loading</p>}

            <table className="">
                <thead>
                    <tr>
                        <th>ID</th>
                        <th>Name</th>
                        <th>State</th>
                        <th>Country</th>
                        <th>Longitude</th>
                        <th>Latitude</th>
                    </tr>
                </thead>
                <tbody>
                    {
                        images.map(img => (
                            <GridItem
                                key={img.id}
                                {...img}
                            />
                        ))
                    }
                </tbody>
            </table>
        </>
    )
}

Grid.propTypes = {
    city: PropTypes.string.isRequired
}