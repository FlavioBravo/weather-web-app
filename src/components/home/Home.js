import React, { useState } from 'react';

import { AddName } from './AddName';
import { Grid } from './Grid';

export const Home = ({ defaultCities = [] }) => {

    const [cities, setCities] = useState( defaultCities );

    return (
      <>
          <h2>Weather Web App</h2>
          <AddName setCities={ setCities } />
          <hr />

          <ol>
              {
                  cities.map( city  => (
                      <Grid 
                          key={ city }
                          city={ city }
                      />
                  ))
              }
          </ol>

      </>
  )
}
