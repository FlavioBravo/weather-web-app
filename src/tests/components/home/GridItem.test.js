import React from 'react';
import { shallow } from 'enzyme';
import { GridItem } from '../../../components/home/GridItem';


describe('Pruebas en <GridItem />', () => {

    const forecast = {
        id: 3690608,
        name: "Zorritos",
        state: "",
        country: "PE",
        coord: {
            lon: -80.678192,
            lat: -3.68046
        }
    };
    const wrapper = shallow( <GridItem { ...forecast } /> );

    
    test('debe de mostrar el componente correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();

    })

    test('debe de tener 6 columnas', () => {
        
        const td = wrapper.find('td');
        expect( td.length ).toBe(6);

    })

    test('debe de tener la clase animate__fadeIn', () => {
        
        const tr = wrapper.find('tr');
        const className = tr.prop('className');

        expect( className.includes('animate__fadeIn') ).toBe( true );

    })

    test('debe de tener el contenido igual a los props', () => {
        
        const tr = wrapper.find('tr');
        expect( tr.text().trim() ).toBe( '3690608  Zorritos    PE  -80.678192  -3.68046' );

    }); 

})