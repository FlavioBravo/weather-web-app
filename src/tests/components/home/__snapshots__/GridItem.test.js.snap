// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GridItem /> debe de mostrar el componente correctamente 1`] = `
<tr
  className="animate__animated animate__fadeIn"
  onClick={[Function]}
>
  <td>
     
    3690608
     
  </td>
  <td>
     
    Zorritos
     
  </td>
  <td>
     
     
  </td>
  <td>
     
    PE
     
  </td>
  <td>
     
    -80.678192
     
  </td>
  <td>
     
    -3.68046
     
  </td>
</tr>
`;
